PennyMe/ViewController.swift

```diff
@@ -146,20 +155,27 @@ class ViewController: UIViewController, UITextFieldDelegate, UIGestureRecognizer
         super.viewWillAppear(animated)
 
         //  Check whether reload has to be triggered
-        if shouldReloadData() {
+        // only triggered if it's not already running
+        if shouldReloadData() && (!isLoadingServerLocations) {
+            PennyMap.removeAnnotations(artworks)
             artworks = Artwork.artworks()
+            isVisible = [:]
             pinIdDict = [:]
             loadInitialData()
+            addAnnotationsIteratively()
         }
         // each time the view appears, check colours of the pins
         check_json_dict()
         // check whether some setting has changed, if yes, reload all data on the map
         if SettingsViewController.hasChanged {
+            updateIncludedStates()
             addAnnotationsIteratively()
             SettingsViewController.hasChanged = false
         }
         if SettingsViewController.clusterHasChanged {
+            // reload all annotations in that case
```

Comment:
We can, but it shouldn't change anything, because clusterHasChanged and hasChanged are distinct (see [here](https://github.com/jannisborn/PennyMe/blob/8767381f63947614e21a13000ced43e019154e71/PennyMe/SettingsViewController.swift#L104))